module-4/nmap/nmap.py

```diff
@@ -0,0 +1,80 @@
+import socket
+import argparse
+import concurrent.futures
+
+# Common services for the first 100 ports
+COMMON_PORTS = {
+    20: "FTP data",
+    21: "FTP control",
+    22: "SSH",
+    23: "Telnet",
+    25: "SMTP",
+    53: "DNS",
+    80: "HTTP",
+    110: "POP3",
+    143: "IMAP",
+    443: "HTTPS",
+    993: "IMAPS",
+    995: "POP3S",
+    # ... continue up to port 100 as needed
+}
+
+def resolve_host(host):
```

Comment:
for better readability, understandability and code-style reasons you should rename this function to something more descriptive, e.g.
`resolve_host_ip_from_name()`